posthog/hogql/printer.py

```diff
@@ -1137,11 +1139,24 @@ def visit_call(self, node: ast.Call):
                     if not has_tz_override:
                         args.append(self.visit(ast.Constant(value=self._get_timezone())))
 
+                    # If the datetime is in correct format, use optimal toDateTime, it's more strict but faster
+                    # and it allows CH to use index efficiently.
+                    if (
+                        relevant_clickhouse_name == "parseDateTime64BestEffortOrNull"
+                        and len(node.args) == 1
+                        and isinstance(node.args[0].type, StringType)
```

Comment:
Might be worth checking `                        and isinstance(node.args[0], ConstantType)`

Reply:
add: `and isinstance(node.args[0], Constant)`